apex/contrib/test/fused_dense/test_fused_dense.py

```diff
@@ -10,39 +12,50 @@
 
 
 @unittest.skipIf(SKIP_TEST, f"{SKIP_TEST}")
-class FusedDenseTest(unittest.TestCase):
-    def setUp(self, seed=0):
-        torch.manual_seed(seed)
+class FusedDenseTest(common_utils.TestCase):
 
-        self.seq_length   = 512
-        self.sequences    = 3
-        self.hidden_dim   = 1024
+    def _test_fused_dense(self, dtype, seed=0):
 
-        self.ref_inputs = torch.randn(self.sequences*self.seq_length, self.hidden_dim,
-                                      dtype=torch.float16, device=torch.device("cuda")).int().half().requires_grad_(True)
+        os.environ["TORCH_ALLOW_TF32_CUBLAS_OVERRIDE"] = "0"
+        torch.manual_seed(seed)
 
-        self.tst_inputs = self.ref_inputs.clone().detach().requires_grad_(True)
-        self.dense = fused_dense.FusedDense(1024, 3072)
-        self.dense.half()
-        self.dense.cuda()
+        seq_length = 512
+        sequences = 3
+        hidden_dim = 1024
 
+        ref_inputs = torch.randn(sequences*seq_length, hidden_dim,
+                                 dtype=dtype, device=torch.device("cuda")).int().to(dtype=dtype).requires_grad_(True)
```

Comment:
why casting to int then cast back to dtype?

Reply:
Good point, the original code did this but it doesn't make sense to me either, I'll get rid of the extra casts.